src/modules/assets/assets.subscriber.ts

```diff
@@ -130,8 +157,54 @@ export class AssetsEventSubscriber {
     const saved = await this.historyRepository.save(eventToSave);
     this.logger.debug(
       `${eventToSave.type} event for ${
-        eventToSave.assetId
+      eventToSave.assetId
       } handled: ${JSON.stringify(saved)}`,
     );
+
+    const msg = {
+      event: MessageType.AssetOfferRejected,
+      asset: event.assetId,
+      rejectedBy: event.emittedBy,
+      offeredBy: event.relatedTo,
+      at: event.at
+    };
+    this.nats.connection.publish(
+      `${msg.offeredBy}.${NATS_EXCHANGE_TOPIC}`,
+      JSON.stringify(msg)
+    );
+  }
+  
+  @OnEvent(AssetHistoryEventType.ASSET_OFFER_CANCELED)
+  async handleAssetOfferCanceled(event: AssetEvent) {
+    if (
+      await this.isEventAlreadySaved(
+        AssetHistoryEventType.ASSET_OFFER_CANCELED,
+        event,
+      )
+    ) {
+      return;
+    }
+    const eventToSave = AssetsHistory.create({
+      ...event,
+      type: AssetHistoryEventType.ASSET_OFFER_CANCELED,
+    });
+    const saved = await this.historyRepository.save(eventToSave);
+    this.logger.debug(
+      `${eventToSave.type} event for ${
+      eventToSave.assetId
+      } handled: ${JSON.stringify(saved)}`,
```

Comment:
```suggestion
      } saved: ${JSON.stringify(saved)}`,
```
Minor point, but I would use "saved" instead of "handled" here as technically the "handled" function is executed even if `isEventAlreadySaved == true`